src/views/preview/preview.jsx

```diff
@@ -97,6 +103,38 @@ class Preview extends React.Component {
     removeEventListeners () {
         window.removeEventListener('popstate', this.handlePopState);
     }
+    getExtensions (projectId) {
+        storage
+            .load(storage.AssetType.Project, projectId, storage.DataFormat.JSON)
+            .then(projectAsset => {
+                let input = projectAsset.data;
+                if (typeof input === 'object' && !(input instanceof ArrayBuffer) &&
+                !ArrayBuffer.isView(input)) { // taken from scratch-vm
+                    // If the input is an object and not any ArrayBuffer
+                    // or an ArrayBuffer view (this includes all typed arrays and DataViews)
+                    // turn the object into a JSON string, because we suspect
+                    // this is a project.json as an object
+                    // validate expects a string or buffer as input
+                    // TODO not sure if we need to check that it also isn't a data view
+                    input = JSON.stringify(input);
+                }
+                parser(projectAsset.data, false, (err, projectData) => {
+                    if (err) {
+                        console.log('Caught project parsing error:', err); // eslint-disable-line no-console
```

Comment:
Instead of `console.log`, use `lib/log`, e.g.: https://github.com/LLK/scratch-www/blob/417791b589fd5dca0616d6e112811c8c279587ca/src/components/errorboundary/errorboundary.jsx#L18